File: common/src/main/java/DahRealPanda/plugins/colonyweb/colony/model/EquipmentInfo.java
package DahRealPanda.plugins.colonyweb.colony.model;

/**
 * One equipped item — a piece of armour or a held weapon.
 *
 * <p>{@link #armorPoints} is what decides whether one guard is better protected than another,
 * and is what the roster sorts on.</p>
 */
public class EquipmentInfo extends ItemInfo {
    public String slot;        // "Head", "Chest", "Legs", "Feet", "Main hand", "Off hand"
    public int armorPoints;    // vanilla armour value, 0 for weapons and tools
    public boolean enchanted;
    public int durabilityPct = 100; // 100 when undamaged or unbreakable
}

File: common/src/main/kotlin/DahRealPanda/plugins/colonyweb/model/EquipmentInfo.kt
package DahRealPanda.plugins.colonyweb.model

/**
 * One equipped item — a piece of armour or a held weapon.
 *
 * [armorPoints] is what decides whether one guard is better protected than another,
 * and is what the roster sorts on.
 */
data class EquipmentInfo(
    @JvmField var slot: String = "",        // "Head", "Chest", "Legs", "Feet", "Main hand", "Off hand"
    @JvmField var armorPoints: Int = 0,     // vanilla armour value, 0 for weapons and tools
    @JvmField var enchanted: Boolean = false,
    @JvmField var durabilityPct: Int = 100  // 100 when undamaged or unbreakable
) : ItemInfo()
